### PR TITLE
Fix unicode encode error when generating slugs

### PR DESCRIPTION
**Pull request by karenc**
When baking College Physics on qa.cnx.org:

```
Traceback (most recent call last):
  File "/var/cnx/venvs/publishing/local/lib/python2.7/site-packages/celery/app/trace.py", line 382, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/tasks.py", line 31, in __call__
    return super(PyramidAwareTask, self).__call__(*args, **kwargs)
  File "/var/cnx/venvs/publishing/local/lib/python2.7/site-packages/celery/app/trace.py", line 641, in __protected_call__
    return self.run(*args, **kwargs)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/db.py", line 74, in wrapped
    return func(*args, **kwargs)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/subscribers.py", line 167, in baking_processor
    bake(binder, recipe_id, publisher, message, cursor=cursor)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/db.py", line 70, in wrapped
    return func(*args, **kwargs)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/bake.py", line 92, in bake
    amend_tree_with_slugs(tree)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/utils.py", line 70, in amend_tree_with_slugs
    amend_tree_with_slugs(node, title_seq)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/utils.py", line 70, in amend_tree_with_slugs
    amend_tree_with_slugs(node, title_seq)
  File "/var/cnx/venvs/publishing/src/cnx-publishing/cnxpublishing/utils.py", line 67, in amend_tree_with_slugs
    tree['slug'] = generate_slug(*title_seq)
  File "/var/cnx/venvs/publishing/local/lib/python2.7/site-packages/cnxcommon/urlslug.py", line 47, in generate_slug
    result = '{}-{}'.format(number, result)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 32: ordinal not in range(128)
```

This is because College Physics contains unicode apostrophes in some of
its chapters and sections.

### CODE REVIEW

File: cnxcommon/urlslug.py [hunk 1]
```diff
@@ -51,10 +53,14 @@ def generate_slug(book_title, *other_titles):
...
 
 def remove_html_tags(title):
+    if isinstance(title, bytes):
+        title = title.decode('utf-8')
     return re.sub(r"<.*?>", "", title)
 
 
 def get_os_number(title):
+    if isinstance(title, bytes):
+        title = title.decode('utf-8')
```
Comment (mmulich) [inline cnxcommon/urlslug.py]: How about a decorator? 😸

File: tests/test_urlslug.py [hunk 2]
```diff
@@ -188,3 +188,24 @@ def test_removes_apostrophes(self):
...
+
+        slug = generate_slug(*titles)
+        assert slug == '5-introduction-further-applications-of-newtons-laws'
+
+        titles = (
+            u'College Physics',
+            u'<span class="os-number">5</span><span class="os-divider"> </span><span class="os-text">'
+            u'Further Applications of Newton’s Laws: Friction, Drag, and Elasticity</span>',
+            u'<span class="os-text">Introduction: Further Applications of Newton’s Laws</span>',
+        )
```
Comment (mmulich) [inline tests/test_urlslug.py:209]: Can you split this test into two distinct test functions? And also say that this second one is specifically test the handling of unicode in the parent title?

**Review by mmulich — CHANGES_REQUESTED** [review]
Suggesting some changes for clarity.

File: tests/test_urlslug.py (hunk 2 above)
Comment (karenc) [reply tests/test_urlslug.py:209]: Actually the two tests have the same text, just one has the inputs as bytes and the other one as unicode.  I'll split them into different functions.

File: tests/test_urlslug.py [hunk 3]
```diff
@@ -188,3 +188,25 @@ def test_removes_apostrophes(self):
...
+            u'Further Applications of Newton’s Laws: Friction, Drag, and Elasticity</span>'.encode('utf-8'),
+            u'<span class="os-text">Introduction: Further Applications of Newton’s Laws</span>'.encode('utf-8')
+        )
+
+        slug = generate_slug(*titles)
+        assert slug == '5-introduction-further-applications-of-newtons-laws'
+
+    def test_handling_bytes(self):
+        titles = (
+            u'College Physics',
```
Comment (m1yag1) [inline tests/test_urlslug.py:205]: am i seeing this correctly? It says `test_handling_bytes` yet I see the each of the titles begins with unicode. The same goes w/ the one above. It `says test_handling_unicode` yet has bytes set at the beginning of the title for `College Physics`.

Comment (karenc) [reply tests/test_urlslug.py:205]: @m1yag1 Yes, it's testing that we can handle `bytes` (the python type) that has unicode in them.

Comment (karenc) [reply tests/test_urlslug.py:205]: :rofl: Ok, you're right, the test names should be the other way round.  Good catch!

Comment (karenc) [reply tests/test_urlslug.py:205]: Updated!

**Review by therealmarv — APPROVED** [review]
from my side now :+1:

**Review by m1yag1 — APPROVED** [review]

**Review by mmulich — APPROVED** [review]
